integration/test/performancetest/transmitter.go

```diff
@@ -153,12 +197,182 @@ func (transmitter *TransmitterAPI) TableExist() (bool, error) {
+	newPacket: this is the agentData collected in this test
+	currentPacket: this is the agentData stored in dynamo currently
+*/
+func (transmitter * TransmitterAPI) PacketMerger(newPacket map[string]interface{}, currentPacket map[string]interface{}) (map[string]types.AttributeValue,error){
+	testSettings := fmt.Sprintf("%s-%s",os.Getenv(PERFORMANCE_NUMBER_OF_LOGS),os.Getenv(TPS))
+	fmt.Println("The test is",testSettings)
+	item := currentPacket["Results"].(map[string]interface{})
+	_,isPresent := item[testSettings] // check if we already had this test
+	if isPresent{ 
+		// we already had this test so ignore it
+		return nil,errors.New("Nothing to update")
+	}
+	testSettingValue:=newPacket["Results"].(map[string]map[string]Stats)[testSettings]
+	mergedResults := make(map[string]map[string]interface{})
+	mergedResults["Results"] = make(map[string]interface{})
+	for attribute,value := range item{
+		_, isPresent := newPacket["Results"].(map[string]map[string]Stats)[attribute]
+		if(isPresent){continue}
+		mergedResults["Results"][attribute] = value
+		
+	}
+	
+	mergedResults["Results"][testSettings] = testSettingValue
+	newAttributes, _ := attributevalue.MarshalMap(mergedResults)
+	return newAttributes, nil
+}
+/*
+UpdateItem()
+Desc:
+	This function updates the item in dynamo if the atomic condition is true else it will return ConditionalCheckFailedException 
+Params:
+	hash: this is the commitHash
+	targetAttributes: this is the targetAttribute to be added to the dynamo item
+	testHash: this is the hash of the last item, used like a version check
+*/
+func (transmitter * TransmitterAPI) UpdateItem(packet map[string]interface{}) error{
+	var ae *types.ConditionalCheckFailedException // this exception represent the atomic check has failed
+	rand.Seed(time.Now().UnixNano())
+	randomSleepDuration := time.Duration(rand.Intn(UPDATE_DELAY_THRESHOLD))*time.Second
+	hash:= packet[HASH].(string)
+	for attemptCount:=0; attemptCount< MAX_ATTEMPTS; attemptCount++{
+		fmt.Println("Updating:",hash)
+		item,err := transmitter.Query(hash) // get most Up to date item from dynamo | O(1) bcs of global sec. idx.
+		if len(item) ==0{ // check if hash is in dynamo
+			return errors.New("ERROR: Hash is not found in dynamo")
+		}
+		commitDate := fmt.Sprintf("%d",int(item[0][COMMIT_DATE].(float64)))
+		year := fmt.Sprintf("%d",int(item[0][PARTITION_KEY].(float64)))
+		testHash := item[0][TEST_ID].(string)
+		targetAttributes,err := transmitter.PacketMerger(packet,item[0])
+		if err !=nil{
+			return err
+		}
+		//setup the update expression
+		expressionAttributeValues := make(map[string]types.AttributeValue)
+		expression := ""
+		n_expression := len(targetAttributes)
+		i :=0
+		for attribute, value := range targetAttributes{
+			expressionName := ":" +strings.ToLower(attribute)
+			expression += fmt.Sprintf("set %s = %s",attribute,expressionName)
+			expressionAttributeValues[expressionName] = value
+			if(n_expression -1 >i){
+				expression += "and"
+			}
+			i++
```

Comment:
BTW you can use `strings.Join` to concatenate a bunch of strings.
```go
exprs := []string{"a", "b", "c"}
res := strings.Join(exprs, " and ")
```
https://go.dev/play/p/I3mTZzbyQ_6

I don't really mind how it's written now since the issue is that it's a map, but wanted to bring it up in case you didn't know that the `Join` function exists.